### PR TITLE
plerkle: remove

### PR DESCRIPTION
**Pull request by fanatid**
Remove plerkle serialization structs and use structs from solana creates, see https://github.com/metaplex-foundation/digital-asset-rpc-infrastructure/pull/144

### CODE REVIEW

**Comment by kespinola** [comment]
https://github.com/metaplex-foundation/blockbuster/issues/39

plerkle_serialization is currently a dev-dependency. This issue drafted to track its removal. Overall like to move to fixture based test suite. It has that now but also some helper logic for generating random instructions that I believe swapped with more fixtures.

File: blockbuster/Cargo.toml [hunk 1]
```diff
@@ -1,34 +1,34 @@
 [package]
 name = "blockbuster"
 description = "Metaplex canonical program parsers, for indexing, analytics etc...."
-version = "1.0.1"
```
Comment (kespinola) [inline blockbuster/Cargo.toml:4]: Using minor let me know if we want to publish as patch.

File: blockbuster/src/instruction.rs [hunk 2]
```diff
@@ -30,123 +28,71 @@ impl<'a> Default for InstructionBundle<'a> {
...
-            } else {
-                // legacy no stack height list must exist if no compiled or no processing will be done
-                let inner_instructions = legacy_inner_ix_list.unwrap();
-                inner_instructions
+    for (outer_instruction_index, message_instruction) in message_instructions.iter().enumerate() {
+        let non_hoisted_inner_instruction = meta_inner_instructions
+            .iter()
+            .filter(|ix| ix.index == outer_instruction_index as u8)
+            .flat_map(|ix| {
+                ix.instructions
```
[the remark this answers is not in the document]
Comment (kespinola) [reply blockbuster/src/instruction.rs]: I still owe Pedro from Helius an issue to discuss providing more structure to the transaction bundle processed by das. We can use stack height to put together logic structure for the transaction representation where its denormalized.

Something like the below but we can fully represent the parent child relationship of the instructions nesting cpi calls  within the instruction that invoked.
 
```
pub struct Instruction {
    pub program_id: Pubkey,
    pub data: Vec<u8>,
}

pub struct InstructionGroup {
    pub outer_instruction: Instruction,
    pub inner_instructions: Vec<Instruction>,
}

pub struct TransactionInfo {
    pub slot: Slot,
    pub block_time: Option<UnixTimestamp>,
    pub instruction_groups: Vec<InstructionGroup>,
} 
```

File: blockbuster/src/instruction.rs [hunk 3]
```diff
@@ -30,123 +28,71 @@ impl<'a> Default for InstructionBundle<'a> {
...
-                                })
-                                .collect::<Vec<IxPair>>()
-                        } else {
-                            Vec::new()
-                        }
-                    })
-                    .collect::<Vec<IxPair>>()
-            } else {
-                // legacy no stack height list must exist if no compiled or no processing will be done
-                let inner_instructions = legacy_inner_ix_list.unwrap();
```
[the remark this answers is not in the document]
Comment (fanatid) [reply blockbuster/src/instruction.rs:83]: pre-1.16?

`compiled_inner_instructions` / `let inner_instructions = legacy_inner_ix_list.unwrap();` moved to plerkle deserialization into solana structs

[the remark this answers is not in the document]
Comment (danenbm) [reply blockbuster/src/instruction.rs:83]: ok ty

**Review by danenbm — APPROVED** [review]

**Pull request merged**